src/main/java/mate/academy/bookstore/service/BookServiceImpl.java

```diff
@@ -25,16 +26,38 @@ public BookDto save(CreateBookRequestDto requestDto) {
 
     @Override
     public List<BookDto> getAll() {
-        return bookRepository.getAll().stream()
+        return bookRepository.findAll().stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
     @Override
     public BookDto getBookById(Long id) {
-        Book book = bookRepository.getBookById(id).orElseThrow(
+        Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException(CANNOT_GET_BOOK_BY_ID_EXCEPTION + id)
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        Optional<Book> bookById = bookRepository.findById(id);
+        if (bookById.isPresent()) {
+            Book updatedBook = new Book();
+            updatedBook.setId(id);
+            updatedBook.setAuthor(bookRequestDto.getAuthor());
+            updatedBook.setTitle(bookRequestDto.getTitle());
+            updatedBook.setPrice(bookRequestDto.getPrice());
+            updatedBook.setIsbn(bookRequestDto.getIsbn());
+            updatedBook.setDescription(bookRequestDto.getDescription());
+            updatedBook.setCoverImage(bookRequestDto.getCoverImage());
+            return bookMapper.toDto(bookRepository.save(updatedBook));
+        }
```

Comment:
Simplify it, please. In this method you don't need to set manually all params. Instead, you already have this params in bookRequestDto, because before you request body on contoller layer. Check does this id exist, and you can declare Book and with bookMapper just map bookRequestDto to book, after manually set your id, which you get in method input params, and in the end convert back your book with id to dto and return it.